pkg/operator/main.go

```diff
@@ -151,7 +153,7 @@ func main() {
 	}
 
 	// initialize the GameServerBuild controller
-	if err = controllers.NewGameServerBuildReconciler(mgr, portRegistry).SetupWithManager(mgr); err != nil {
+	if err = controllers.NewGameServerBuildReconciler(mgr, portRegistry, int(cfg.MaxNumberOfGameServersToAdd), int(cfg.MaxNumberOfGameServersToDelete)).SetupWithManager(mgr); err != nil {
```

Comment:
let's modify the NewGameServerBuildReconciler method to accept the entire cfg struct as argument, since in the future we might want to grab more parameters from there.

Reply:
Hey @dgkanatsios 

Thanks for the feedback! I was going to incorporate this change, but it looks like I'm not able to use the Config struct within gameserverbuild_controller.go since the Config struct is defined in main.go

The workaround I can try out is moving the struct out of main.go and putting it in its own file (similar to port registry) but I'm not sure if the configs will still pull the env variables properly. It also would require the suite_test.go to have a slight overhaul. Let me know what you think!

Reply:
> The workaround I can try out is moving the struct out of main.go and putting it in its own file

this sounds like a great solution, let’s try this! Thank you!

Reply:
Hey @dgkanatsios 

I may need some help. E2e tests are passing but any unit tests related to gameserverbuild are failing, I'm thinking I need to modify something with the gameserverbuild_types.go but nothing really sticks out to me. Unit test failures also aren't showing me any reasons outside of timeouts, even after bumping timeout to 5 mins.

Reply:
make sure the kind cluster is deleted before you run unit tests locally (or just stop docker). There is a port conflict if this happens. I thought we had written that on the instructions somewhere :(

Reply:
Right, I've been running them with the kind cluster deleted. The PR's unit tests are also failing.

Reply:
Workaround found and implemented.